internal/ecsservicediscovery/decoratedtask.go

```diff
@@ -138,6 +142,23 @@ func (t *DecoratedTask) generatePrometheusTarget(
 	addExporterLabels(labels, taskGroupLabel, t.Task.Group)
 	addExporterLabels(labels, taskStartedbyLabel, t.Task.StartedBy)
 	addExporterLabels(labels, taskLaunchTypeLabel, t.Task.LaunchType)
+
+	if arn, err := arn.Parse(*t.Task.TaskArn); err == nil {
+		// ARN formats: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-account-settings.html#ecs-resource-ids
+		splitResource := strings.Split(arn.Resource, "/")[1:]
```

Comment:
My _only_ hesitation with this is if there were some drastic change and ARNs were changed to have zero `/` characters, this would panic, IIRC. But that feels like an impossible situation to me. And if it were to happen, I imagine such a breaking change would have a long runway for adoption, so we would have (hopefully) ample time to accommodate for that. Cross that bridge when we get there.

Reply:
I'll still sleep at night even if this doesn't get changed, but I would feel better if there was a check like
```go
splitResource := strings.Split(arn.Resource, "/")
numTokens = len(splitResource)
if numTokens >= 3 {
  // assume index 1 and 2 are the task ID and cluster name
  clusterName := splitResource[1]
  taskId := splitResource[2]
} else if numTokens == 2 {
  taskId := splitResource[1]
}
```

Reply:
I had the same concern, but Go handles it nicely: https://go.dev/play/p/EFjwYRYb7tZ

Reply:
Ah okay. That works for me. My bad for assuming it would panic instead of just giving an empty splice.